### DISCUSSION TITLE
JK BMS & LAN board

### DISCUSSION THREAD

**Discussion by ghost** [Sleeper85/esphome-jk-bms-can#14] [Show and tell]
Hi Sleeper. 
I just want to share , that your  CAN version and also BLE version works with ESP32 board version with LAN port. I am using Olimex Gateway rev. G board . It works using DHCP and also with manual IP set in YAML.
 It works for me with following settings 

# GPIO pins your CAN bus transceiver (TJA1050, TJA1051T or SN65HVD230) is connected to the ESP32 TX->TX and RX->RX !
  can_tx_pin: GPIO33
  can_rx_pin: GPIO34
# GPIO pins your JK-BMS UART-TTL is connected to the ESP32 TX->RX and RX->TX !
  tx_pin: GPIO4
  rx_pin: GPIO32

I tried to use GPIO 12&13 , but it doesn't work for me . Lan doesn't get up. It is written in manual for Olimex , that it is free to use, but look like it is not true. 

Comment or delete wifi section and substitute with following code
ethernet:
  type: LAN8720
  mdc_pin: GPIO23
  mdio_pin: GPIO18
  clk_mode: GPIO17_OUT
  phy_addr: 0

if anybody  would like to use manual IP , set it in DHCP server or put in YAML following code.

  # Optional manual IP
  manual_ip:
    static_ip: your IP 
    gateway: your gateway
    subnet: your subnet

**Comment by Sleeper85** [comment]
Hi,

Ok, thanks for your feedback.

Based on the example file below. Can you prepare a new file for this board?

https://github.com/Sleeper85/esphome-jk-bms-can/blob/main/config/config_generic-esp32-devkit-v1.yaml

I will then add it to the list of compatible boards.

Thanks
